R/PipeOpLearnerCV.R

```diff
@@ -113,29 +119,29 @@
 PipeOpLearnerCV = R6Class("PipeOpLearnerCV",
   inherit = PipeOpTaskPreproc,
   public = list(
-    initialize = function(learner, id = NULL, param_vals = list()) {
+    initialize = function(learner, resampling = rsmp("cv", folds = 3), id = NULL, param_vals = list()) {
       private$.learner = as_learner(learner, clone = TRUE)
       private$.learner$param_set$set_id = ""
-      id = id %??% private$.learner$id
-      # FIXME: can be changed when mlr-org/mlr3#470 has an answer
+      private$.resampling = as_resampling(resampling, clone = TRUE)
+      private$.resampling$param_set$set_id = "resampling"
+
+      # tags of resampling parameters should include "train"; we fix this here
+      for (i in seq_along(private$.resampling$param_set$params)) {
+        private$.resampling$param_set$params[[i]]$tags = c("train", private$.resampling$param_set$params[[i]]$tags)
+      }
+
+
+      id = id %??% self$learner$id
       task_type = mlr_reflections$task_types[get("type") == private$.learner$task_type][order(get("package"))][1L]$task
 
-      private$.crossval_param_set = ParamSet$new(params = list(
-        ParamFct$new("method", levels = c("cv", "insample"), tags = c("train", "required")),
-        ParamInt$new("folds", lower = 2L, upper = Inf, tags = c("train", "required")),
+      private$.additional_param_set = ParamSet$new(params = list(
         ParamLgl$new("keep_response", tags = c("train", "required"))
       ))
-      private$.crossval_param_set$values = list(method = "cv", folds = 3, keep_response = FALSE)
-      private$.crossval_param_set$set_id = "resampling"
-      # Dependencies in paradox have been broken from the start and this is known since at least a year:
-      # https://github.com/mlr-org/paradox/issues/216
-      # The following would make it _impossible_ to set "method" to "insample", because then "folds"
-      # is both _required_ (required tag above) and at the same time must be unset (because of this
-      # dependency). We will opt for the least annoying behaviour here and just not use dependencies
-      # in PipeOp ParamSets.
-      # private$.crossval_param_set$add_dep("folds", "method", CondEqual$new("cv"))  # don't do this.
+      private$.additional_param_set$values = list(keep_response = FALSE)
```

Comment:
name `resampling.keep_response` (and hope no resampling method has `keep_response` as parameter)